sqldelight-gradle-plugin/src/main/kotlin/app/cash/sqldelight/gradle/GenerateMigrationOutputTask.kt

```diff
@@ -76,6 +78,7 @@ abstract class GenerateMigrationOutputTask : SqlDelightWorkerTask() {
         properties = properties,
         verifyMigrations = false,
         compilationUnit = parameters.compilationUnit.get(),
+        dialect = ServiceLoader.load(SqlDelightDialect::class.java).findFirst().get(),
```

Comment:
```suggestion
        dialect = ServiceLoader.load(SqlDelightDialect::class.java).single(),
```
I think this is what you want since you want to throw if there's more than one available. Otherwise you can still replace `findFirst().get()` with Kotlin's `first()` for the same effect.

Below too.